driver-core/src/main/com/mongodb/client/model/expressions/ArrayExpression.java

```diff
@@ -61,4 +63,29 @@
      */
     T reduce(T initialValue, BinaryOperator<T> in);
 
+    IntegerExpression size();
+
+    T elementAt(IntegerExpression i);
+
+    default T elementAt(final int i) {
+        return this.elementAt(of(i));
+    }
+
+    T first();
+
+    T last();
+
+    BooleanExpression contains(T contains);
+
+    ArrayExpression<T> concat(ArrayExpression<T> array);
+
+    ArrayExpression<T> slice(IntegerExpression start, IntegerExpression length);
```

Comment:
Would that not be `firstN`?

I considered using combinations of skip and first, so for example:

```
first(3) -- slice(0, 3)
skip(5)
skip(5).first(3) -- slice(5, 3)
last(5).first(3) -- slice(-5, 3)
skipLast(1).last(3)
```

This seems verbose, but I personally find the alternatives hard to remember (I believe slice was copied from JS, which also has splice, subarray, substr, substring...). This is also conceptually identical to offset/limit (skip/first), which it seems we will use elsewhere.

Reply:
> Would that not be firstN?

`slice(-n, n)` is `lastN(n)`. But there is no `lastN` in `ArrayExpression`. If we introduce it, that'd even better.

> I considered using combinations of skip and first

There is also no `skip` in `ArrayExpression`.

> the alternatives hard to remember

I agree, it's very hard for me to even read the test code that uses negative values and out of bounds values when testing `slice`. So can we have `firstN`/`lastN`?

Reply:
I will not implement these methods now, since they are outside of the "top 50". But please leave a comment on https://github.com/mongodb/mongo-java-driver/pull/1032 and I will see if there is time to add them later. They are not quite trivial to add.